### PR TITLE
Decouple with a mocked service

### PR DESCRIPTION
**Pull request by tian-lt**
## What
The azure resource for mocked currency data is going to be deprecated. We can't rely on that endpoint to sustain our development builds.

## How
1. Removed `DataLoaderMockConstants.h`
1. Refactored `CurrencyHttpClient`
1. Provides an override implementation for `CurrencyHttpClient` in `CalcViewModeCopyForUT`
1. Update currency unit tests accordingly.
1. Removed `ICurrencyHttpClient` since it's an interface for unit testing.

### CODE REVIEW

**Review by guominrui — DISMISSED** [review]
Looks good to me.

**Review by hanzhang54 — APPROVED** [review]

File: src/CalcViewModel/DataLoaders/CurrencyHttpClient.h [hunk 1]
```diff
@@ -2,17 +2,39 @@
...
+        }
+
+        void await_suspend(std::experimental::coroutine_handle<>) const noexcept
+        {
+            assert(false && "not implemented.");
+        }
+
+        T&& await_resume() noexcept
+        {
+            return std::forward<T>(Value);
```
Comment (guominrui) [inline src/CalcViewModel/DataLoaders/CurrencyHttpClient.h:26]: `move`?

Comment (tian-lt) [reply src/CalcViewModel/DataLoaders/CurrencyHttpClient.h]: updated

File: src/CalcViewModel/DataLoaders/CurrencyHttpClient.h [hunk 2]
```diff
@@ -2,17 +2,39 @@
...
+
     class CurrencyHttpClient
     {
     public:
         static bool ForceWebFailure;
         void Initialize(Platform::String ^ sourceCurrencyCode, Platform::String ^ responseLanguage);
 
-        Platform::String ^ GetCurrencyMetadata() const;
-        Platform::String ^ GetCurrencyRatios() const;
+        MockAwaitable<Platform::String ^> GetCurrencyMetadata() const;
```
Comment (guominrui) [inline src/CalcViewModel/DataLoaders/CurrencyHttpClient.h:36]: Suggest `Async` suffix for these two methods.

**Review by hanzhang54 — APPROVED** [review]

**Review by guominrui — APPROVED** [review]

**Pull request merged**